docs/design/gcp-wif-support_design.md

```diff
@@ -0,0 +1,155 @@
+`listener-design.md`. -->
+
+## Abstract
+Support Google Cloud's WIF (Workload Identity Federation) for OADP.
+
+## Background
+In currently released versions of OADP, the only way to authenticate to GCP is via a service account.
+This is not ideal for customers who are using GCP's WIF feature to authenticate to GCP.
+This proposal aims to add support for WIF to OADP.
+
+## Goals
+- GCP WIF support for OADP and Velero for backup and restore of applications backed by GCP resources.
+- Using OpenShift's Cloud Credentials Operator to generate a short-lived token for authentication to GCP.
+
+## Non Goals
+- ImageStreamTag backup and restore (follow up design/implementation)
+- [Standardized update flow for OLM-managed operators leveraging short-lived token authentication](https://issues.redhat.com/browse/OCPSTRAT-95) (follow up design/implementation)
+- Allowing customers to use another long lived tokens separate from the one used by the Cloud Credentials Operator to generate short-lived tokens.
+
+
+## High-Level Design
+<!-- One to two paragraphs that describe the high level changes that will be made to implement this proposal. -->
+A wiki will be made available to customers to follow the steps to configure GCP WIF for OADP. The wiki will also include steps to configure the Cloud Credentials Operator to generate a short-lived token for authentication to GCP. Updates Velero GCP plugin to use the short-lived token for authentication to GCP.
+
+## Detailed Design
+
+
+### Prerequisites
+- Cluster installed in manual mode [with GCP Workload Identity configured](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.12/html-single/authentication_and_authorization/index#gcp-workload-identity-mode-installing).
+    - This means you should now have access to `ccoctl` CLI from this step and access to associated workload-identity-pool.
+
+### Create Credential Request for OADP Operator
+- Create oadp-credrequest dir
+    ```bash
+    mkdir -p oadp-credrequest
+    ```
+- Create credrequest.yaml
+    ```bash
+    echo 'apiVersion: cloudcredential.openshift.io/v1
+    kind: CredentialsRequest
+    metadata:
+      name: oadp-operator-credentials
+      namespace: openshift-cloud-credential-operator
+    spec:
+      providerSpec:
+        apiVersion: cloudcredential.openshift.io/v1
+        kind: GCPProviderSpec
+        permissions:
+        - compute.disks.get
+        - compute.disks.create
+        - compute.disks.createSnapshot
+        - compute.snapshots.get
+        - compute.snapshots.create
+        - compute.snapshots.useReadOnly
+        - compute.snapshots.delete
+        - compute.zones.get
+        - storage.objects.create
+        - storage.objects.delete
+        - storage.objects.get
+        - storage.objects.list
+        - iam.serviceAccounts.signBlob
+        skipServiceCheck: true
+      secretRef:
+        name: cloud-credentials-gcp
+        namespace: openshift-adp
+      serviceAccountNames:
+      - velero
```

Comment:
Shouldn't this be operator SA `openshift-adp-controller-manager` ?

Reply:
nah. this is for velero SA to use.openshift-adp-controller-manager sa do not actually access or create data to bucket/snapshots.

Reply:
oh ok, makes sense.